pkg/shares/parse_sparse_shares.go

```diff
@@ -31,6 +31,14 @@ func parseSparseShares(rawShares [][]byte, supportedShareVersions []uint8) (blob
 			return nil, fmt.Errorf("unsupported share version %v is not present in supported share versions %v", version, supportedShareVersions)
 		}
 
+		isNamespacedPadded, err := IsNamespacedPadded(share)
+		if err != nil {
+			return nil, err
+		}
+		if isNamespacedPadded {
+			continue
+		}
```

Comment:
[question]
what is the current mechanism for skipping namespace padding shares? if there is an old mechanism, do we need to remove it? is there a way to test that this is actually getting hit? do we even need such a test?

Reply:
Great question. We do have such a unit test here: https://github.com/rootulp/celestia-app/blob/bf8026570fada9b2a638f38a81d1f012bad45c2a/pkg/shares/parse_sparse_shares_test.go#L131 which continues to pass.

Previously, namespaced padded shares were added to the blob preceeding them and then trimmed from the previous blob [here](https://github.com/rootulp/celestia-app/blob/bf8026570fada9b2a638f38a81d1f012bad45c2a/pkg/shares/parse_sparse_shares.go#L78-L80). We don't want to remove those lines because a partial blob share will still have padding at the end of the share and those lines are responsible for trimming the trailing `0s`.

Reply:
nice!